snapcraft/plugins/ruby.py

```diff
@@ -0,0 +1,160 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 James Beedy <jamesbeedy@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The ruby plugin is useful for ruby based parts.
+The plugin uses gem to install dependencies from a `Gemfile`.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keywords:
+    - gems:
+      (list)
+      A list of gems to install.
+    - use-bundler
+      (boolean)
+      Use bundler to install gems from a Gemfile (defaults 'false').
+    - ruby-version:
+      (string)
+      The version of ruby you want this snap to run.
+"""
+import os
+import re
+import platform
+
+from snapcraft import BasePlugin, file_utils
+from snapcraft.internal.errors import SnapcraftEnvironmentError
+from snapcraft.sources import Tar
+
+
+class RubyPlugin(BasePlugin):
+
+    @classmethod
+    def schema(cls):
+        schema = super().schema()
+
+        schema['properties']['use-bundler'] = {
+            'type': 'boolean',
+            'default': 'false'
+        }
+        schema['properties']['ruby-version'] = {
+            'type': 'string',
+            'default': '2.3.1'
+        }
+        schema['properties']['gems'] = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {
+                'type': 'string'
+            },
+            'default': []
+        }
+        return schema
+
+    @classmethod
+    def get_pull_properties(cls):
+        # Inform Snapcraft of the properties associated with pulling. If these
+        # change in the YAML Snapcraft will consider the build step dirty.
+        return ['ruby-version', 'gems']
+
+    def __init__(self, name, options, project):
```

Comment:
I'd like to tag this plugin as being "in beta" so we can change its options a bit while it gets some use. Would you mind adding a warning (using logger.warn) to the __init__ saying something like "The ruby plugin is currently in beta, its API may break. Use at your own risk".